src/core/test/system/system.t.sol

```diff
@@ -321,4 +324,23 @@ contract SystemTest is TestUtils, DSTest {
         borrower.doBorrow(loan, 100);
         assertEq(currency.balanceOf(borrower_), 100);
     }
+
+    function testSupply() public {
+        supply(100 ether, 10 ether);
+        assertEq(currency.balanceOf(juniorInvestor_), 90 ether);
+        assertEq(lenderDeployer.juniorERC20().balanceOf(juniorInvestor_), 10 ether);
+        assertEq(currency.balanceOf(address(borrowerDeployer.shelf())), 10 ether);
+    }
+
+    function testRedeem() public {
+        supply(100 ether, 10 ether);
+        assertEq(currency.balanceOf(juniorInvestor_), 90 ether);
+        assertEq(lenderDeployer.juniorERC20().balanceOf(juniorInvestor_), 10 ether);
+        // additional liquidity changes the token price
+        // TODO: add code to move funds from shelf to tranche so we can enact the actual repayment scenario
+        currency.mint(address(lenderDeployer.junior()), 100 ether);
+        juniorInvestor.doRedeem(10 ether);
+        assertEq(lenderDeployer.juniorERC20().balanceOf(juniorInvestor_), 0);
+        assertEq(currency.balanceOf(juniorInvestor_), 190 ether);
```

Comment:
Shouldn't it be `200 ether`

I think the test is not working because you need to change the flag in the distributor to repay.